file_formats/cmt_format.ml

```diff
@@ -197,12 +213,169 @@ let clear_env binary_annots =
 
   else binary_annots
 
+let iter_on_usages ~index =
+  let f ~namespace env path lid =
+    let not_ghost { Location.loc = { loc_ghost; _ }; _ } = not loc_ghost in
+    if not_ghost lid then
+      try
+        let shape = Env.shape_of_path ~namespace env path in
+        let shape = Local_reduce.weak_reduce env shape in
+        if not (Shape.is_closed shape) then
+          index := (Unresolved shape, lid) :: !index
+        else Option.iter
+          (fun uid -> index := (Resolved uid, lid) :: !index)
+          shape.Shape.uid
+      with Not_found -> ()
+  in
+  let add_constructor_description env lid =
+    function
+    | { Types.cstr_tag = Cstr_extension (path, _); _ } ->
+        f ~namespace:Extension_constructor env path lid
+    | { Types.cstr_uid = Predef "()"; _ } ->
+        (* We do not index occurrences of `()` *)
```

Comment:
Why is there a special treatment for `()`?.. but not for the type `unit`, or everything else that is "predefined"?